testsuites/CBLTester/CBL_Functional_tests/APITests/test_dictionary.py

```diff
@@ -0,0 +1,265 @@
+        value = "Test String"
+        for i in range(num_of_keys):
+            hashmap["{}_{}".format(key, i)] = value
+        content_dict = self.dict_obj.create(hashmap)
+        assert num_of_keys == self.dict_obj.count(content_dict)
+
+    @pytest.mark.parametrize("key, value", [
+        (random_string(6), True),
+        (random_string(6), False)
+    ])
+    def test_get_set_boolean(self, key, value):
+        '''
+        @summary: Testing get and set Boolean methods of Dictionary API
+        '''
+        content_dict = self.dict_obj.create()
+        self.dict_obj.setBoolean(content_dict, key, value)
+        assert self.dict_obj.getBoolean(content_dict, key) == value
+
+    def test_get_set_date(self):
+        '''
+        @summary: Testing get and set Date methods of Dictionary API
+        '''
+        # TODO implementation does not work on ios
+        key = "Date_key"
+        value = self.datatype.setDate()
+        content_dict = self.dict_obj.create()
+        self.dict_obj.setDate(content_dict, key, value)
+        # assert self.datatype.compare(value,
+        #                            self.dict_obj.getDate(content_dict,
+        #                                                   key))
+        assert self.datatype.compareDate(value, self.dict_obj.getDate(content_dict, key))
+
+    def test1_get_set_dictionary(self):
+        '''
+        @summary: Testing get and set Dictionary methods of Dictionary API
+        '''
+        # TODO ios gets back {} for self.datatype.hashMap()
+        hashmap = {}
+        key = "Date_key"
+        value = self.datatype.setDate()
+        hashmap[key] = value
+        key = "Double_key"
+        value = self.datatype.setDouble(2.0)
+        hashmap[key] = value
+        key = "Float_key"
+        value = self.datatype.setFloat(3.0)
+        hashmap[key] = value
+        key = "Integer_key"
+        value = 4
+        hashmap[key] = value
+        key = "Long_key"
+        value = self.datatype.setLong(1234567890)
+        hashmap[key] = value
+        key = "String_key"
+        value = "Test String"
+        hashmap[key] = value
+        content = self.dict_obj.create(hashmap)
+        content_dict = self.dict_obj.create()
+        self.dict_obj.setDictionary(content_dict, "hashmap", content)
+        content_check = self.dict_obj.getDictionary(content_dict, "hashmap")
+        for key in hashmap:
+            assert self.dict_obj.contains(content_check, key)
+
+    @pytest.mark.parametrize("key, value", [
+        (random_string(6), "{}".format(random.uniform(0, 1))),
+        (random_string(6), "{}".format(random.uniform(1, 10))),
+        (random_string(6), "{}".format(random.uniform(11, 100))),
+        (random_string(6), "{}".format(random.uniform(101, 1000)))
+    ])
+    def test_get_set_double(self, key, value):
+        '''
+        @summary: Testing get and set Double methods of Dictionary API
+        '''
+        # TODO implementation does not work on ios
```

Comment:
What needs to be done here?

Reply:
Fixes already done. Removed TODO comments